### PR TITLE
Change the behavior of field_is_complex to be more similar to pydantic v1

### PR DESCRIPTION
**Pull request by dmontagu**
I discovered this discrepancy while helping Sebastian with FastAPI. @hramezani I'll defer to you whether the change makes sense. Feel free to modify the implementation, add tests, etc. Or request that I do it 😄

### CODE REVIEW

**Review by samuelcolvin — COMMENTED** [review]
i think the behaviour is fine, but we should keep the method on the source so uses can override it more easily.

File: pydantic_settings/sources.py [hunk 1]
```diff
@@ -4,15 +4,16 @@
...
 from pathlib import Path
-from typing import TYPE_CHECKING, Any, Dict, List, Mapping, Optional, Tuple, Type, Union
+from typing import TYPE_CHECKING, Any, Dict, List, Mapping, Optional, Sequence, Tuple, Type, Union
 
 from pydantic import BaseModel
 from pydantic._internal._typing_extra import origin_is_union
 from pydantic._internal._utils import deep_update, lenient_issubclass
 from pydantic.fields import FieldInfo
-from typing_extensions import get_origin
+from typing_extensions import get_args, get_origin
```
Comment (hramezani) [inline pydantic_settings/sources.py:16]:
```suggestion
from typing_extensions import get_origin
```

**Review by hramezani — APPROVED** [review]
LGTM 👍

**Comment by hramezani** [comment]
Thanks @dmontagu

**Pull request merged**